pkg/csi/service/cns/controller.go

```diff
@@ -97,6 +103,17 @@ func (c *controller) Init(config *config.Config) error {
 		klog.Errorf("checkAPI failed for vcenter API version: %s, err=%v", vc.Client.ServiceContent.About.ApiVersion, err)
 		return err
 	}
+	// Create the kubernetes client from config
+	k8sclient, err := k8s.NewClient()
+	if err != nil {
+		klog.Errorf("Creating Kubernetes client failed. Err: %v", err)
+		return err
+	}
+	// Set up kubernetes resource listeners for metadata syncer
+	c.k8sInformerManager = k8s.NewInformer(k8sclient)
+	c.pvLister = c.k8sInformerManager.GetPVLister()
+	c.vaLister = c.k8sInformerManager.GetVALister()
+	c.k8sInformerManager.Listen()
 	c.nodeMgr = &Nodes{}
 	err = c.nodeMgr.Initialize()
```

Comment:
`k8s.NewInformer()` is called from this function as well. And it also calls `Listen()` - https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/pkg/csi/service/vanilla/nodes.go#L52

I see that `NewInformer()` is a singleton now, so what happens when the same informer manager calls `Listen()` twice?

Reply:
Yes, with Singleton `InformerManager` ensures that the stop signal is invoked only once. However, listeners will be on a separate go routine: 1 for listening on the nodes and 1 for listening on PV, VA